shared/js/background/classes/tab.es6.js

```diff
@@ -48,15 +48,31 @@ class Tab {
         this.cleanAmpUrl = null
         this.requestId = tabData.requestId
         this.status = tabData.status
-        this.site = new Site(this.url)
+
+        // Consider the site URL to be the request initiator for requests fired
+        // inside opaque 'tabs' (e.g. via ServiceWorkers). Otherwise, consider
+        // the site URL to be the request URL.
+        const requestInitiator = tabData.initiator || tabData.originUrl
```

Comment:
Internally service workers always have a clear origin AFAIK - is our method of determining it reliable? In what cases initiator or originUrl are empty?

<img width="538" alt="sw" src="https://user-images.githubusercontent.com/985504/173554951-e659e83f-889d-47fb-845f-15c9df5bbf05.png">

Reply:
I don''t think it is safe to assume the initiator URL is available, because in my experience it isn't always available. For example, perhaps requests initiated by other places will have a tab ID of -1 but no initiator URL. I figure it's better not to throw an exception in those cases, or to convert `undefined` to a URL of `"undefined"` by accident.